gapic-generator/lib/gapic/schema/api.rb

```diff
@@ -243,8 +244,42 @@ def grpc_service_config
         end
       end
 
+      # Get a resource given its type string
+      def lookup_resource_type resource_type
+        @resource_types[resource_type]
+      end
+
       private
 
+      def analyze_resources
```

Comment:
The description passage from the PR belongs here as a comment to explain what is going on

Reply:
done